### PR TITLE
add compile tests

### PR DESCRIPTION
**Pull request by llogiq**
This adds a number of compile tests to ensure that "wrong" usage patterns actually get caught. Also adds a test for some known good usages (e.g. using two arenas without mixing them up).

### CODE REVIEW

**Comment by llogiq** [comment]
cc @kpp can you confirm that this covers #18?

File: tests/ui/mixup.rs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+extern crate compact_arena;
+
+use compact_arena::in_nano_arena;
+
+fn main() {
+    in_nano_arena!(a, {
+        in_nano_arena!(b, {
+            let _ = a.add(1usize);
```
Comment (kpp) [inline tests/ui/mixup.rs]: Why did you add two entities to `a`?

**Comment by kpp** [comment]
And that example from https://github.com/llogiq/compact_arena/issues/18 is not covered. In tests you used `in_*_arena` pattern, not `mk_*_arena` pattern

File: tests/ui/mixup.rs (hunk 1 above)
Comment (llogiq) [reply tests/ui/mixup.rs]: To really get an out of bounds access.

**Comment by llogiq** [comment]
Ah, I see. I'll add another test.

File: tests/ui/mixup.rs (hunk 1 above)
Comment (kpp) [reply tests/ui/mixup.rs]: But you don't need an out of bounds access, you need compile time error that `x` cannot be used as an index of `b`
